test/async.coffee

```diff
@@ -204,3 +204,21 @@ test "#3199: await multiline implicit object", ->
         type: 'a'
         msg: 'b'
     eq undefined, y
+
+test "top-level await", ->
+  eq CoffeeScript.compile('await null'), '''
+    (async function() {
+      await null;
+
+    }).call(this);
+
+  '''
+  eq CoffeeScript.compile('-> await null'), '''
+    (function() {
+      (async function() {
+        return (await null);
+      });
+
+    }).call(this);
+
+  '''
```

Comment:
I would change this to be a bit more realistic:

```suggestion
  eqJS 'do -> await null', '''
    (async function() {
      return (await null);
    })();
  '''
```

But it doesn’t belong in a “top level await” section, as this `await` isn’t top-level.

Reply:
What I meant to test here is that the wrapper is *not* marked `async` (in non-bare mode). I can give this test a better (and separate) name though.